Decory edits to documentation example notebooks

dedwar65 opened the pull request:
Please ensure your pull request adheres to the following guidelines:

<!--- Put an `x` in all the boxes that apply: -->
- [ ] Tests for new functionality/models or Tests to reproduce the bug-fix in code.
- [ ] Updated documentation of features that add new functionality.
- [ ] Update CHANGELOG.md with major/minor changes.

llorracc commented:
@dedwar65 the tests are showing that there is something wrong with the 

examples/Journeys/Journey-PhD.ipynb

notebook.  Could you open it up on your computer and verify that it works, then `git add . ` and `git commit -m '[message]'` and `git push`

dedwar65 commented:
Just took a look and tried some initial fixes. The error message reads “Unexpected token < in JSON at position 10323”

Any suggestions on where I can learn about JSON errors of this type specific to the notebook (other than google)?



I’ve tried some using of double quotation marks and such but not much luck resolving the issue.



It seems like the only real change from the notebook should’ve been the naming of the file right?



Decory

From: Christopher Llorracc Carroll ***@***.***>
Date: Sunday, January 15, 2023 at 1:37 PM
To: econ-ark/HARK ***@***.***>
Cc: Decory Edwards ***@***.***>, Mention ***@***.***>
Subject: Re: [econ-ark/HARK] Decory edits to documentation example notebooks (PR #1215)

      External Email - Use Caution





@dedwar65<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdedwar65&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=HId7Lbg6wfRgmTge7QeE3nPUeAozTLq1bQKpikN4elI%3D&reserved=0> the tests are showing that there is something wrong with the

examples/Journeys/Journey-PhD.ipynb

notebook. Could you open it up on your computer and verify that it works, then git add . and git commit -m '[message]' and git push

—
Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fecon-ark%2FHARK%2Fpull%2F1215%23issuecomment-1383221173&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=CrPlVDYtnOhpgQnGk9CRKd9VheAruaKDFoVIRuwPvMY%3D&reserved=0>, or unsubscribe<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAR3YDSOU7UI5DQEWOYHQGZ3WSQ7X7ANCNFSM6AAAAAAT3P6BTE&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AOF9AI2jTDcQozILjoJml8%2Fwc%2FkiAP3ezmuQ1hJTnzc%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

llorracc commented:
Does the version in the copy  ("HARK-merged-bad" or whatever you called it)
of the repo work?

On Sun, Jan 15, 2023 at 2:38 PM Decory Edwards ***@***.***>
wrote:

> Just took a look and tried some initial fixes. The error message reads
> “Unexpected token < in JSON at position 10323”
>
> Any suggestions on where I can learn about JSON errors of this type
> specific to the notebook (other than google)?
>
>
>
> I’ve tried some using of double quotation marks and such but not much luck
> resolving the issue.
>
>
>
> It seems like the only real change from the notebook should’ve been the
> naming of the file right?
>
>
>
> Decory
>
> From: Christopher Llorracc Carroll ***@***.***>
> Date: Sunday, January 15, 2023 at 1:37 PM
> To: econ-ark/HARK ***@***.***>
> Cc: Decory Edwards ***@***.***>, Mention ***@***.***>
> Subject: Re: [econ-ark/HARK] Decory edits to documentation example
> notebooks (PR #1215)
>
> External Email - Use Caution
>
>
>
>
>
> @dedwar65<
> https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdedwar65&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=HId7Lbg6wfRgmTge7QeE3nPUeAozTLq1bQKpikN4elI%3D&reserved=0>
> the tests are showing that there is something wrong with the
>
> examples/Journeys/Journey-PhD.ipynb
>
> notebook. Could you open it up on your computer and verify that it works,
> then git add . and git commit -m '[message]' and git push
>
> —
> Reply to this email directly, view it on GitHub<
> https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fecon-ark%2FHARK%2Fpull%2F1215%23issuecomment-1383221173&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=CrPlVDYtnOhpgQnGk9CRKd9VheAruaKDFoVIRuwPvMY%3D&reserved=0>,
> or unsubscribe<
> https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAR3YDSOU7UI5DQEWOYHQGZ3WSQ7X7ANCNFSM6AAAAAAT3P6BTE&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AOF9AI2jTDcQozILjoJml8%2Fwc%2FkiAP3ezmuQ1hJTnzc%3D&reserved=0
> >.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/econ-ark/HARK/pull/1215#issuecomment-1383233431>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKCK74FXPTEVUPVZ6QBPWDWSRG2FANCNFSM6AAAAAAT3P6BTE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
- Chris Carroll

dedwar65 commented:
Actually it doesn’t :/ I tried the Journey notebook from both repos. I am wondering if I should just remake It from jupyter notebook, but there has to be a quicker way right?

Decory

From: Christopher Llorracc Carroll ***@***.***>
Date: Sunday, January 15, 2023 at 5:22 PM
To: econ-ark/HARK ***@***.***>
Cc: Decory Edwards ***@***.***>, Mention ***@***.***>
Subject: Re: [econ-ark/HARK] Decory edits to documentation example notebooks (PR #1215)

      External Email - Use Caution




Does the version in the copy ("HARK-merged-bad" or whatever you called it)
of the repo work?

On Sun, Jan 15, 2023 at 2:38 PM Decory Edwards ***@***.***>
wrote:

> Just took a look and tried some initial fixes. The error message reads
> “Unexpected token < in JSON at position 10323”
>
> Any suggestions on where I can learn about JSON errors of this type
> specific to the notebook (other than google)?
>
>
>
> I’ve tried some using of double quotation marks and such but not much luck
> resolving the issue.
>
>
>
> It seems like the only real change from the notebook should’ve been the
> naming of the file right?
>
>
>
> Decory
>
> From: Christopher Llorracc Carroll ***@***.***>
> Date: Sunday, January 15, 2023 at 1:37 PM
> To: econ-ark/HARK ***@***.***>
> Cc: Decory Edwards ***@***.***>, Mention ***@***.***>
> Subject: Re: [econ-ark/HARK] Decory edits to documentation example
> notebooks (PR #1215)
>
> External Email - Use Caution
>
>
>
>
>
> @dedwar65<
> https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdedwar65&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=HId7Lbg6wfRgmTge7QeE3nPUeAozTLq1bQKpikN4elI%3D&reserved=0>
> the tests are showing that there is something wrong with the
>
> examples/Journeys/Journey-PhD.ipynb
>
> notebook. Could you open it up on your computer and verify that it works,
> then git add . and git commit -m '[message]' and git push
>
> —
> Reply to this email directly, view it on GitHub<
> https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fecon-ark%2FHARK%2Fpull%2F1215%23issuecomment-1383221173&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=CrPlVDYtnOhpgQnGk9CRKd9VheAruaKDFoVIRuwPvMY%3D&reserved=0>,
> or unsubscribe<
> https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAR3YDSOU7UI5DQEWOYHQGZ3WSQ7X7ANCNFSM6AAAAAAT3P6BTE&data=05%7C01%7Cdedwar65%40jhu.edu%7C9f48f983f1234134adad08daf7279cd8%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094046763300411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AOF9AI2jTDcQozILjoJml8%2Fwc%2FkiAP3ezmuQ1hJTnzc%3D&reserved=0
> >.
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/econ-ark/HARK/pull/1215#issuecomment-1383233431>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKCK74FXPTEVUPVZ6QBPWDWSRG2FANCNFSM6AAAAAAT3P6BTE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


--
- Chris Carroll


—
Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fecon-ark%2FHARK%2Fpull%2F1215%23issuecomment-1383268835&data=05%7C01%7Cdedwar65%40jhu.edu%7Cb360ef5d26064553be3b08daf7470199%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094181597220693%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=lfCP5wAU9Eq4Koy5cxOt9MEpDn%2BvDceVPMaiJevgoRQ%3D&reserved=0>, or unsubscribe<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAR3YDSMD2UENRP6VSO3OXJTWSR2CVANCNFSM6AAAAAAT3P6BTE&data=05%7C01%7Cdedwar65%40jhu.edu%7Cb360ef5d26064553be3b08daf7470199%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094181597220693%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=d1sGnHh392alV8dKlPnZoUNws1SzqB81TVzyvG82pwA%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

llorracc commented:
@dedwar65,

I've tried to fix it.  The problem was that the journey doc was corrupted by the incomplete attempt to merge it.  I removed the bad parts and hope it will now work.  

The lesson: Merge conflicts are very delicate to solve.  Whenever you get the message about "the document has been changed in both [the remote branch] and [your branch]" it may be best to just 

1. make a local copy of the document that you have edited
1. replace your local version with the upstream version of the file
1. do a `git status` to make sure that now there is no longer a conflict
1. `push` your local version (to make sure it is properly sync'd
1. Use `meld` to carefully compare your copy of the file to the version you have retrieved from upstream
    * Copy the changes you have made and want to keep into the version from upstream
1. git add, git commit, and git push

dedwar65 commented:
Thanks for helping with this. I will save these steps for future use.

I do have another question: It says that I am 3 commits ahead of the master version. These should be the notebooks that I removed examples from, the cycles_tutorial notebook I added, and the addition of the ConsPortfolio dictionary for the portfolio example notebook on the documentation.

But I thought I already put in a pull request for these changes? When I go to “hark.readthedocs.io”, the changes don’t seem to be there, but they do show up on my local version. Do I need to make another pull request?

Decory

From: Christopher Llorracc Carroll ***@***.***>
Date: Sunday, January 15, 2023 at 6:05 PM
To: econ-ark/HARK ***@***.***>
Cc: Decory Edwards ***@***.***>, Mention ***@***.***>
Subject: Re: [econ-ark/HARK] Decory edits to documentation example notebooks (PR #1215)

      External Email - Use Caution





@dedwar65<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdedwar65&data=05%7C01%7Cdedwar65%40jhu.edu%7C3b606dca901a42f099b808daf74d0b63%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094207544720804%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=aJmSrtSVbe49h1IYUNuXK6fNugRx0JGSYUmmqi%2F9mhw%3D&reserved=0>,

I've tried to fix it. The problem was that the journey doc was corrupted by the incomplete attempt to merge it. I removed the bad parts and hope it will now work.

The lesson: Merge conflicts are very delicate to solve. Whenever you get the message about "the document has been changed in both [the remote branch] and [your branch]" it may be best to just

  1.  make a local copy of the document that you have edited
  2.  replace your local version with the upstream version of the file
  3.  do a git status to make sure that now there is no longer a conflict
  4.  push your local version (to make sure it is properly sync'd
  5.  Use meld to carefully compare your copy of the file to the version you have retrieved from upstream

     *   Copy the changes you have made and want to keep into the version from upstream

  1.  git add, git commit, and git push

—
Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fecon-ark%2FHARK%2Fpull%2F1215%23issuecomment-1383277652&data=05%7C01%7Cdedwar65%40jhu.edu%7C3b606dca901a42f099b808daf74d0b63%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094207544720804%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=0PCKcQ9Y78EYjnu1VRG4Urxg%2ByO04YnG%2BpvaU0jYUu8%3D&reserved=0>, or unsubscribe<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAR3YDSJOA2UFEA5HUET2KQTWSR7E3ANCNFSM6AAAAAAT3P6BTE&data=05%7C01%7Cdedwar65%40jhu.edu%7C3b606dca901a42f099b808daf74d0b63%7C9fa4f438b1e6473b803f86f8aedf0dec%7C0%7C0%7C638094207544877042%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=VkWN%2FJT6KfXxao7dC%2BeSpw9KfO3sG2rc%2BQBsuest5uQ%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
